### PR TITLE
greatis/Anti-WebMiner

### PR DESCRIPTION
**Pull request by ParaplegicRacehorse**
Undoubtedly some overlap with other lists. Also likely some hosts not included in any of the other lists. I did not thoroughly check this against other lists for uniqueness.

Also, linked directly to [greatis/Anti-WebMiner/](https://github.com/greatis/Anti-WebMiner) `hosts` file.
Link used without contacting @greatis.
And no idea how up-to-date file is going to be kept.

---
**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`

`{Please ensure that your pull request is for the 'development' branch!}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [] I have tested my proposed changes, and have included unit tests where possible.
- [] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))

---

**What does this PR aim to accomplish?:**
additional hosts to block web-based cryptocurrency miners.

**How does this PR accomplish the above?:**
links to hosts file maintained in another GH repo - [greatis/Anti-WebMiner](https://github.com/greatis/Anti-WebMiner)

**What documentation changes (if any) are needed to support this PR?:**
none (?)

### CODE REVIEW

**Comment by Mcat12** [comment]
Why should the default lists include this list? Pi-hole is not targeted as being a web-miner blocker.

**Comment by ParaplegicRacehorse** [comment]
Some web-mining sites are already blocked in at least one of the default lists, but as to why block web-miners:

1. Ads are annoying but do not, essentially, use significant amounts of compute power or network bandwidth.
2. Pi-Hole already blocks lots of call-home services which, again, are annoying but generally do not consume a significant amount of compute power or network bandwidth.
3. Web (js) mining software does not consume a significant amount of network bandwidth but DOES use an enormous amount of compute power; enough to significantly slow down the computer's user-experience.
4. Most of these sites are really shady and don't tell you they're _stealing_ your computer's resources (and your electrical power for which you pay the bill) to make them "money."
5. Some of these sites embed the miner in a cookie-like structure which is kept by the browser and continues to run even after the browser-tab is closed or page moved away from so long as the browser is open and, if not flushing cookies every session, restarts when the browser restarts.

In short: these web-miners are extremely badly behaved, even compared to the worst call-home and advertising services.

If the devs would rather, I'll cancel this pull request and simply add an issue feature-request for Pi-Hole to include by default more anti-webmining simply because of how insidiously evil these things are.

**Comment by dschaper** [comment]
We already make it quite easy to add custom lists to the Pi-hole installation. In order to include a list as default it would need the authors consent as we would be pulling their lists very often. We've been burned in the past with adding lists to default that turned up to be not capable of handling the task, or became over aggressive and blocked too much. I share your view as to the sites to be blocked, but I think with the ability for users to add lists themselves that we need to be conservative in our defaults.

**Comment by ParaplegicRacehorse** [comment]
@dschaper, you are correct. I just thought it would be best to have an anti-cryptominer file in the default configs. I would volunteer to make such a file available on a fast, high-bandwidth VPS host (or, even, contained in this repository as part of the base distribution) but I am at sea for months at a time and am unable to effectively maintain such a list.

I consider the PR closed.

Please do, however, add a suggestion in the settings for users to add such a file with a link to one or more good files. Thanks.

**Comment by Mcat12** [comment]
@ParaplegicRacehorse Have you seen @WaLLy3K's list collection? https://wally3k.github.io/

**Comment by ParaplegicRacehorse** [comment]
I had not. Perusing now. Thanks @Mcat12
